### ISSUE TITLE
Modal pages in Android overlaps Android status bar

### ISSUE THREAD

**Issue by david-maw** [dotnet/maui#6214]
### Description

Opening a page using Navigation.PushModalAsync seems to overlap the Android status bar (the very topmost line) rather than stopping just below as it does on Windows.

### Steps to Reproduce

1. Unzip the project from [MultiPage-androidmodal.zip](https://github.com/microsoft/microsoft-ui-xaml/files/8509659/MultiPage-androidmodal.zip)
2. Compile and run it on Windows
3. Run it on Android

You should see this:
![multipage-androidmodal-initial](https://user-images.githubusercontent.com/12423046/163917448-a99da35a-71fb-4193-bd8c-2565acde2bcc.jpg)

Then click the Navigation.PopModalAsync button and note that after it switches briefly back to MainPage the same page is presented but covering more of the screen. Now you should see this:
![multipage-androidmodal](https://user-images.githubusercontent.com/12423046/163916167-ef297b49-1acc-405b-b35f-5de7b66f853e.jpg)

Note how the page extends further up on Android than Windows so the status bar is on top of it.

### Version with bug

Release Candidate 1 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android, I was *not* able test on other platforms

### Affected platform versions

Android 11

### Did you find any workaround?

No

**Comment by v-longmin** [comment]
Verified repro on Android 12- API30 with VS 17.2.0 Preview 2.1 [32317.152]. Repro project:
[MultiPage-androidmodal.zip](https://github.com/dotnet/maui/files/8510275/MultiPage-androidmodal.zip)

**Comment by PureWeen** [comment]
@v-longmin can you reverify?

**Comment by david-maw** [comment]
Still seems the same on 17.5 preview 1, am I missing something? 

Also, I don't want to seem rude but having gone to the trouble of creating a reproducible test case to help you diagnose your bug it seems that asking me to reverify it on the latest release is a bit much. If you'd reverified, thought it was fixed and wanted me to verify that I agreed that's one thing, but this seems like you are simply transferring work to me.

On the off chance the problem is the format of the repro i provided I've copied it to a public GitHub repro:
https://github.com/david-maw/MultiPage-AndroidModal.git for your convenience.
